.github/workflows/pytest.yaml

```diff
@@ -7,29 +7,28 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9]
 
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pipenv flake8 wheel
-      - id: cache-pipenv
+      - id: cache-poetry
         uses: actions/cache@v1
         with:
-          path: ~/.local/share/virtualenvs
-          key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
-      - name: Install dependencies
-        if: steps.cache-pipenv.outputs.cache-hit != 'true'
-        run: |
-          pipenv install --deploy --dev
+          path: ~/.local
+          key: ${{ runner.os }}-python-${{ matrix.python-version }}-poetry-${{ hashFiles('**/poetry.lock') }}
```

Comment:
Would it be better for the caching step to be run before the dependencies are installed, so we can check if there's a cached venv or not? And if it does, then load that instead of having to install all dependencies again?

```
      #----------------------------------------------
      #       load cached venv if cache exists      
      #----------------------------------------------
      - name: Load cached venv
        id: cached-poetry-dependencies
        uses: actions/cache@v2
        with:
          path: ~/.local
          key: ${{ runner.os }}-python-${{ matrix.python-version }}-poetry-${{ hashFiles('**/poetry.lock') }}
```

Reply:
Sujay: Yup, that makes sense! Fixed in https://github.com/cancerDHC/example-data/commit/41522b9f898fd516bbf65ad11a97f520c1b1e602.

Mark:
> Is this PR primarily about getting poetry to work with the GH actions? Is the ability to use poetry in a locally cloned repo a direct side effect of that? A new user would still have to install the poetry application and run the poetry install command before doing any local development or testing. right?

It's primarily about getting Poetry settings in here so that you can run the local scripts and tests and things with `poetry run pytest ...`, but one way of testing that we've set that up correctly is that to ensure that GitHub Actions can run the tests via Poetry as well.

This caching is allows us to [keep a copy of the installed dependencies on the GitHub Actions system](https://github.com/snok/install-poetry#caching-the-poetry-installation) -- that way, we don't need to redownload the dependencies every time we run the tests.